repo_lib/__init__.py

```diff
@@ -31,49 +31,53 @@ def __init__(self, path: str, release: str, pool: str,
         self.version = version
         self.origin = origin
 
-    def _archive_cmd(self, command: str, input: str, arch: str = ''):
+    def _archive_cmd(self, command: str, input: str, arch: str = '') -> str:
         cwd = os.getcwd()
         os.chdir(self.path)
         com = ['apt-ftparchive', command, input]
         if arch:
             com.insert(1, f'--arch={arch}')
         output = subprocess.run(['apt-ftparchive', command, input], text=True)
         os.chdir(cwd)
-        return output
+        return output.stdout
 
     def index(self, component: str, arch: str):
         component_dir = join(self.pool, component)
         if not exists(join(self.path, component_dir)):
             raise RepoError('component does not exist',
                             join(self.path, component_dir))
 
-        output_dir = (f'{self.path}/dists/{self.release}'
-                      f'{component}/binary-{arch}')
+        output_dir = join(self.path, 'dists', self.release,
+                          component, f'binary-{arch}')
+
         if not exists(output_dir):
             os.makedirs(output_dir)
 
         output = self._archive_cmd('packages', component_dir, arch=arch)
         with open(join(output_dir, 'Packages'), "w") as fob:
+            if not output:
+                output=""
             fob.write(output)
             if output:
                 fob.write('\n')
```

Comment:
Very petty but this *might* be more succinct if the previous condition was inverted and appended the newline

e.g.

```python3
if not output:
    output = ""
    
fob.write(output)

if output:
    fob.write('\n')
```

becomes

```python3
if output:
    output += '\n'
fob.write(output)
```

Reply:
Nice!